### ISSUE TITLE
Polymorphic responses

### ISSUE THREAD

**Issue by jmeas** [jmeas/redux-resource#270]
Some backends may return polymorphic arrays, such as:

```
[someBook, someAuthor, someComment, someBook]
```

redux-resource's built-in reducer does not handle this at all, and I don't see any easy way to add support for it due to the way that `resourceName` is used to identify every resource within `resources`. There would be a way to write a plugin for it, but losing out on the built-in request statuses plugin is pretty substantial.

This will likely need to be an ecosystem plugin that re-implements the request status plugin in a more general way.

---

Another approach would be a built-in solution like this:

```
{
  resourceName: ['books', 'authors', 'comments'],
  resources: [someBook, someAuthor, someComment, anotherBook]
}
```

but I'm not sure if I want to bloat the API in this way. Additionally, lists gets hairy in this situation.

---

One way to support this would be a lower-level reducing behavior that the other reducing behavior is sugar for.

**Comment by jmeas** [comment]
I'm going to close this as revisit. It's an important question, but it's not a use case that anyone I know of has right now. If your backend has polymorphic responses, and you need a solution, just leave a message here and I'd be happy to chat about it with you!

**Comment by jmeas** [comment]
Reopening as this relates to what I am considering for #297 .

tl;dr,

I am thinking specifying your response in a number of different ways. Right now, it’s like this:

```js
{
  resourceName: ‘books’,
  resources: [{}, {}, {}]
}
```

but it could also be:

```js
{
  books: [{}, {}, {}],
  movies: [{}, {}, {}]
}
```

or:

```js
{
  favoriteBooks: {
    resourceName: ‘books’,
    resources: [{}, {}, {}]
  },
  popularAuthors: {
    resourceName: ‘authors’,
    resources: [{}, {}, {}, {}]
  }
}
```

or!

```js
{
  authorsAndBooks: [
    {
      resourceName: ‘books’,
      ...bookInfo
    },
    {
      resourceName: ‘authors’,
      ...authorInfo
    },
    {
       resourceName: ‘books’,
       ...bookTwoInfo
    }
  ]
}
```

**Comment by jmeas** [comment]
For now, I am going to hold off on this. The reason is because `lists` are still scoped by resource, and polymorphic responses will require polymorphic lists.

I am not opposed to making this breaking change, but it is not high enough impact to justify doing right now. Closing once again for now 🙂
